### PR TITLE
feat: add data truncations to services

### PR DESCRIPTION
**Pull request by CIJones11**
## Description
DTOSS-3343
Add truncation rules to transformRules.json
- FirstName
- Surname
- OtherGivenNames
- PreviousSurname
- AddressLine
- Telephone Number
- Mobile Number
- Email Address

Unit Tests completed for each rules in TransformDataServiceTests.cs

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by Warren-Pitterson — DISMISSED** [review]

**Comment by Warren-Pitterson** [comment]
Approved but needs file format fix on your tests please

File: tests/TransformDataServiceTests/TransformDataServiceTests.cs [hunk 1]
```diff
@@ -168,6 +167,420 @@ public async Task Run_NamePrefixTooLong_TruncatePrefix()
...
+            NhsNumber = "1",
+            FirstName = expectedFirstName,
+            Surname = "Smith",
+            NamePrefix = "MR",
+            Gender = Model.Enums.Gender.Male
+        };
+
+        result.Body.Position = 0; // Reset stream position to beginning
+        var reader = new StreamReader(result.Body, Encoding.UTF8);
+        var responseBody = await reader.ReadToEndAsync();
```
Comment (andrewmorris43) [inline tests/TransformDataServiceTests/TransformDataServiceTests.cs:195]: There is a helper method that could be used here instead using these 3 lines in each test. 

Have a look at the `AssertionHelper.cs` file, there is a method in there called `ReadResponseBodyAsync` that reads the response body.

Comment (CIJones11) [reply tests/TransformDataServiceTests/TransformDataServiceTests.cs]: Updated to use the AssertionHelper ReadResponseBodyAsync method instead

**Review by andrewmorris43 — APPROVED** [review]
Looks good to me.

**Review by andrewmorris43 — APPROVED** [review]

**Pull request merged**